lib/Event/ReorderEvent.php

```diff
@@ -30,7 +32,7 @@ class ReorderEvent extends AbstractDocumentEvent
      */
     public function __construct($document, $destId, $after)
     {
-        parent::__construct($document);
+        $this->document = $document;
```

Comment:
Actually you are using the polymorphic, since you have type hinted the `handleHydrate` method parameter with `AbstractMappingEvent` sometimes, so that you can pass a `PersistEvent` in there. Therefore it would totally make sense to me, to introduce these traits and interfaces, and I guess I will do that in one of the next iterations. Would make some code reuse between these elements also easier, e.g. there is no need to implement `setNode` multiple times.